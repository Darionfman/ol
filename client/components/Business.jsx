import React from 'react'
import * as bs from 'react-bootstrap'
import colectionStore from '../store'
import {observer} from 'mobx-react'

@observer
export default class Business extends React.Component {
  constructor (props) {
    super(props)
  }
  componentDidMount(){
    colectionStore.getBusiness(this.props.id)
  }
  render(){
    const businessData = colectionStore.business
    if(!businessData){
      return (<div>Loading</div>)
    }
    return (
      <bs.Col md={12} className='center'> 
        <h2>{businessData.name}</h2>
        <bs.Col md={12}>
          <a href={`tel:${businessData.phone}`}>
            {businessData.phone}
          </a>
        </bs.Col>
        <bs.Col md={12}>
          <a href={businessData.website} target='_blank' >
            {businessData.website}
          </a>
        </bs.Col>
        <bs.Col md={12}>
          <bs.Col md={12}>
            {`${businessData.address} ${(businessData.address2 || "")}\n${businessData.city} ${businessData.state} ${businessData.country}`}
          </bs.Col>
        </bs.Col>
      </bs.Col>
      )
  }
}
module.exports = Business