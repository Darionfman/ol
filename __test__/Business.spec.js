import React from 'react'
import Business from '../client/components/Business'
import Store from '../client/store'
import renderer from 'react-test-renderer'

test('Business renders', async () => {
  await Store.fetchFirstPage()
  
  const component = renderer.create(
    <Business />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})