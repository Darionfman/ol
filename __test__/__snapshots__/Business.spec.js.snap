exports[`test Business renders 1`] = `
<div
  className="center col-md-12">
  <h2>
    Yundt-Flatley
  </h2>
  <div
    className="col-md-12">
    <a
      className="phone-number"
      href="tel:4034880719">
      4034880719
    </a>
  </div>
  <div
    className="col-md-12">
    <a
      href="http://www.halvorson.com/"
      target="_blank">
      http://www.halvorson.com/
    </a>
  </div>
  <div
    className="col-md-12">
    <div
      className="col-md-12">
      1386 Lim Brooks Suite 517
      Lake Betsy IA US
    </div>
  </div>
</div>
`;
